### ISSUE TITLE
undefined reference to symbol 'hwloc_bitmap_alloc' in netloc_mpi_find_hosts

### ISSUE THREAD

**Issue by leoyanghw** [open-mpi/hwloc#281]
I followed the README installing instructions,
1. ./autogen.sh
2. ./configure
3. make
but error happened as follows:
```
/usr/bin/ld: netloc_mpi_find_hosts-netloc_mpi_find_hosts.o: undefined reference to symbol 'hwloc_bitmap_alloc'
//home/ysk/git_trunk/hwloc/hwloc/.libs/libhwloc.so.0: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
Makefile:512: recipe for target 'netloc_mpi_find_hosts' failed
make[2]: *** [netloc_mpi_find_hosts] Error 1
make[2]: Leaving directory '/home/ysk/git_trunk/hwloc/utils/netloc/mpi'
Makefile:453: recipe for target 'all-recursive' failed
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory '/home/ysk/git_trunk/hwloc/utils'
Makefile:627: recipe for target 'all-recursive' failed
make: *** [all-recursive] Error 1
```
And this is the configure output:
```
Hwloc optional build support status (more details can be found above):

Probe / display I/O devices: PCI(linux) LinuxIO
Graphical output (Cairo):    yes
XML input / output:          basic
Netloc functionality:        yes (without scotch)
Plugin support:              no

```

My environment:
Ubuntu 16.04.4
gcc 5.4.0
autoconf 2.69
automake 1.15
libtool 2.4.6

**Comment by bgoglin** [comment]
Can you post your config.log and the command-line that causes the error (run make V=1 for more verbose output) ?
Also please post the output of 'dpkg -l | grep gcc' and 'dpkg -l | grep binutils' so that I compare with my Ubuntu jenkins slave where things build fine.

**Comment by leoyanghw** [comment]
@bgoglin 
I'm sorry but my proxy rule don't allow me to upload the config.log file. Which sections do you want to check? I will paste it here. The whole log file is too long.

Following is the command causing the error:
```
mv -f .deps/netloc_mpi_find_hosts-netloc_mpi_find_hosts.Tpo .deps/netloc_mpi_find_hosts-netloc_mpi_find_hosts.Po
/bin/bash ../../../libtool  --tag=CC   --mode=link gcc  -g -O2   -o netloc_mpi_find_hosts netloc_mpi_find_hosts-netloc_mpi_find_hosts.o ../../../netloc/libnetloc.la -pthread -Wl,-rpath -Wl,/usr/local/lib -Wl,--enable-new-dtags -L/usr/local/lib -lmpi 
libtool: link: gcc -g -O2 -o .libs/netloc_mpi_find_hosts netloc_mpi_find_hosts-netloc_mpi_find_hosts.o -pthread -Wl,-rpath -Wl,/usr/local/lib -Wl,--enable-new-dtags  ../../../netloc/.libs/libnetloc.so -L/usr/local/lib /usr/local/lib/libmpi.so -pthread
/usr/bin/ld: netloc_mpi_find_hosts-netloc_mpi_find_hosts.o: undefined reference to symbol 'hwloc_bitmap_alloc'
//home/ysk/git_trunk/hwloc/hwloc/.libs/libhwloc.so.0: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
Makefile:512: recipe for target 'netloc_mpi_find_hosts' failed
make[2]: *** [netloc_mpi_find_hosts] Error 1
make[2]: Leaving directory '/home/ysk/git_trunk/hwloc/utils/netloc/mpi'
Makefile:453: recipe for target 'all-recursive' failed
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory '/home/ysk/git_trunk/hwloc/utils'
Makefile:627: recipe for target 'all-recursive' failed
make: *** [all-recursive] Error 1
```
dpkg -l | grep gcc output:
```
ii  gcc                                               4:5.3.1-1ubuntu1                           amd64        GNU C compiler
ii  gcc-5                                             5.4.0-6ubuntu1~16.04.4                     amd64        GNU C compiler
ii  gcc-5-base:amd64                                  5.4.0-6ubuntu1~16.04.4                     amd64        GCC, the GNU Compiler Collection (base package)
ii  gcc-6-base:amd64                                  6.0.1-0ubuntu1                             amd64        GCC, the GNU Compiler Collection (base package)
ii  gir1.2-packagekitglib-1.0                         0.8.17-4ubuntu6~gcc5.4ubuntu1.1            amd64        GObject introspection data for the PackageKit GLib library
ii  libcaca0:amd64                                    0.99.beta19-2build2~gcc5.2                 amd64        colour ASCII art library
ii  libgcc-5-dev:amd64                                5.4.0-6ubuntu1~16.04.4                     amd64        GCC support library (development files)
ii  libgcc1:amd64                                     1:6.0.1-0ubuntu1                           amd64        GCC support library
ii  libpackagekit-glib2-16:amd64                      0.8.17-4ubuntu6~gcc5.4ubuntu1.1            amd64        Library for accessing PackageKit using GLib
ii  libwebrtc-audio-processing-0:amd64                0.1-3ubuntu1~gcc5.1                        amd64        AudioProcessing module from the WebRTC project.
```

'dpkg -l | grep binutils' output:
`ii  binutils                                          2.26.1-1ubuntu1~16.04.3                    amd64        GNU assembler, linker and binary utilities`

**Comment by bgoglin** [comment]
@clementFoyer I guess you need to backport the addition of libhwloc.la to that Makefile.am from your branch.

**Comment by bgoglin** [comment]
@leoyanghw This should be fixed in df29a98467174184a34bd724f316085d770a48d4

By the way, unless you care about netloc, you can pass --disable-netloc at configure time (it will be the default in the upcoming 2.0).
